backend/src/main/java/com/library/backend/controller/MealController.java

```diff
@@ -24,23 +24,50 @@ public ResponseEntity<List<Meal>> getAllMeals() {
     }
 
     @GetMapping("/{id}")
-     public ResponseEntity<Meal> getMealById(@PathVariable String id) {
+     public ResponseEntity<Meal> getMealById(@PathVariable String id) throws Exception {
         Meal meal = mealService.getMealById(id);
         return meal != null ? ResponseEntity.ok(meal) : ResponseEntity.notFound().build();
 
     }
 
     @GetMapping("/category/{category}")
     public ResponseEntity<List<Meal>> getMealsByCategory(@PathVariable String category) {
-        return ResponseEntity.ok(mealService.getMealsByCategory(category));
+        try {
+            return ResponseEntity.ok(mealService.getMealsByCategory(category));
+        } catch (Exception e) {
+            throw new  RuntimeException("Internal Server Error", e);
```

Comment:
Maybe you could have build a custom Exception that tells a bit more about the error that could happen :-)